### PR TITLE
Document development workflow for maintainers

### PR DESCRIPTION
**Pull request by mamg22**
Based on suggestions from https://github.com/nsxiv/nsxiv/discussions/119#discussioncomment-1882806

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -36,3 +36,15 @@ to work on. You can also filter the issues via label:
   Might require some experience.
 * [Help wanted](https://github.com/nsxiv/nsxiv/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22):
   (Intermediate/Experienced) Issues where we require some help.
+
+Development workflow
+--------------------
+
+Our workflow regarding pull requests (PR) is:
```
Comment (N-R-K) [inline CONTRIBUTING.md:43]: This makes it sound like it's applicable to anyone sending a PR when reality it's only for maintainers.

**Review by N-R-K — CHANGES_REQUESTED** [review]

**Comment by N-R-K** [comment]
In the contributions guideline should add one more point:
* Avoid force pushes, especially for bigger patches. Only use them when there's merge conflicts.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -36,3 +38,22 @@ to work on. You can also filter the issues via label:
...
+we may invite you to join the nsxiv org as a member. Being a member simply means
+you will be able to approve, disapprove and merge pull requests. Our workflow
+regarding pull requests is the following:
+
+  * Code related changes require two approvals, but documentation related
+    changes (e.g. typo) can be merged with just one.
+  * Always prefer squashing when merging. In the case a PR makes more than one
+    significant change, use the "don't squash" tag and rebase instead.
+  * When merging, make sure the commit message is cleaned up properly so that
+    it reflects the intention of the PR.
```
Comment (XPhyro) [inline CONTRIBUTING.md:55]:
```suggestion
    it reflects the current intention of the PR.
```

**Review by XPhyro — CHANGES_REQUESTED** [review]

**Comment by XPhyro** [comment]
We should also document the release process:
- The criteria of releasing a new version
- Tagging and release messages

**Comment by N-R-K** [comment]
> * The criteria of releasing a new version
Don't think we have any. Obviously we don't want to tag a release if there's known major issue like compile failure with certain config, but that should be common sense and not really worth documenting.

**Review by XPhyro — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -36,3 +38,44 @@ to work on. You can also filter the issues via label:
...
+    version N".
+  * Update `VERSION` macro in the `Makefile`.
+  * Update the changelog (`CHANGELOG.md`):
+    * Separate each release with three dashes as separators, with a space
+      between each dash.
+    * Include a link with the version name, pointing to the release tarball
+      download, and highlight in bold. In the next line, add the release date,
+      wrapped in parentheses and highlighted in italics.
+    * Document only the changes or fixes between releases.
+    * Don't document internal, non user-visible changes.
```
Comment (N-R-K) [inline CONTRIBUTING.md:74]:
```suggestion
    * Don't document changes which did not make it into a release.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -36,3 +38,44 @@ to work on. You can also filter the issues via label:
...
+thread.
+
+For releases, the process is as follows:
+
+  * Tag the release with a "vN" tag, where N is the version number. Also set
+    the commit message and tag description for the release commit to "Release
+    version N".
+  * Update `VERSION` macro in the `Makefile`.
+  * Update the changelog (`CHANGELOG.md`):
+    * Separate each release with three dashes as separators, with a space
```
Comment (N-R-K) [inline CONTRIBUTING.md:68]: Don't think we need to be so verbose, three dash separator should be obvious.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -36,3 +38,44 @@ to work on. You can also filter the issues via label:
...
+      download, and highlight in bold. In the next line, add the release date,
+      wrapped in parentheses and highlighted in italics.
+    * Document only the changes or fixes between releases.
+    * Don't document internal, non user-visible changes.
+    * Create a list of change sections, each with a sub-list of relevant
+      changes.
+    * Use the "Changes" section to document behavior changes since the last
+      release, the "Added" section for new features, and the "Fixes" section
+      for fixed bugs or regressions.
+    * Include related pull request IDs after each item of the changelog, each
```
Comment (N-R-K) [inline CONTRIBUTING.md:80]: Same as above, I think these last 2 points should be obvious. We should only document non obvious things.

**Review by N-R-K — CHANGES_REQUESTED** [review]

**Review by XPhyro — APPROVED** [review]

**Review by N-R-K — APPROVED** [review]

**Pull request merged**